cmd/server/cmd/server.go

```diff
@@ -258,10 +260,16 @@ func runServer(sa *serverArgs, printf, fatalf shared.FormatFn) {
 		grpcOptions = append(grpcOptions, grpc.Creds(credentials.NewTLS(tlsConfig)))
 	}
 
+	// setup server grpc interceptors for monitoring
+	grpc_prometheus.EnableHandlingTimeHistogram()
+
 	if sa.enableTracing {
 		tracer := bt.New(tracing.IORecorder(os.Stdout))
 		ot.InitGlobalTracer(tracer)
-		grpcOptions = append(grpcOptions, grpc.UnaryInterceptor(otgrpc.OpenTracingServerInterceptor(tracer)))
+		grpcOptions = append(grpcOptions,
+			grpc.UnaryInterceptor(grpc_middleware.ChainUnaryServer(otgrpc.OpenTracingServerInterceptor(tracer), grpc_prometheus.UnaryServerInterceptor)))
+	} else {
+		grpcOptions = append(grpcOptions, grpc.UnaryInterceptor(grpc_prometheus.UnaryServerInterceptor))
```

Comment:
Should we add a flag to control producing prometheus metrics? Then if its true we enable to handling histogram and register the handler.

I didn't realize we already unconditionally register a prom handler in the server too - arguably it could be behind the same flag. I think it'd be reasonable to have the flag default to true, though.